### ISSUE TITLE
simplify experiment_summary.md

### ISSUE THREAD

**Issue by msalganik** [niznik-dev/cruijff_kit#201]
# Describe the Feature

The `experiment_summary.md` file is bloated (unnecessary sections like `Quick Reference` that do not directly cater to humans or machines). We want to:
- [x]  slim down the file generation that takes place in `experiment_generation.md` to only include information that would be utilized in later skills
- [x] convert `experiment_summary.md` to a `.yaml` structure (see [torchtune_config_writer](https://github.com/msalganik/torchtune_config_writer?tab=readme-ov-file)) to increase consistency
- [x] limit variation between files generated from the same user specifications (i.e. `workflow_test.yaml`)
- [x] distinguish between _per experiment_ settings and _per user_ settings (such as Write `chmod` permissions that should most likely be in `CLAUDE.md` in the repo rather than `experiment_summary.md`)
- [x] remove any "Status" logic from the outputted files (should not be rewritten by any later tasks)
- [x] examine whether "Compute Estimates" are necessary (read in scaffolding?) and remove if not

## Related Issues
#164

**Comment by msalganik** [comment]
Here's an example that I created here using this branch: https://github.com/niznik-dev/cruijff_kit/pull/197

# Experiment: Cross-Length Generalization in Capitalization

**Created:** 2025-11-16
**Type:** Research Experiment
**Status:** Designed (ready for scaffolding)

## Overview

**Scientific Question:** Does fine-tuning on word capitalization for one word length generalize to different word lengths?

**Hypothesis:** Models trained on capitalization for one word length may or may not generalize to different word lengths, revealing whether the model learns a general capitalization rule or memorizes length-specific patterns.

**Experimental Design:**
- Train model A on 5-letter words → Evaluate on both 5-letter (in-distribution) AND 9-letter (out-of-distribution) words
- Train model B on 9-letter words → Evaluate on both 5-letter (out-of-distribution) AND 9-letter (in-distribution) words
- Base model control → Evaluate on both word lengths to establish baseline performance

This creates a 3×2 evaluation matrix testing bidirectional generalization between word lengths.

**Total Runs:** 2 fine-tuning runs + 1 base model control = 3 runs
**Total Evaluations:** 6 (3 runs × 2 evaluation datasets)

## Tools

- **Model Preparation:** torchtune
  - *Purpose:* Fine-tuning LLMs with LoRA
  - *Used by:* `scaffold-torchtune` and `run-torchtune` skills

- **Evaluation:** inspect-ai
  - *Purpose:* Evaluating LLMs on custom tasks
  - *Used by:* `scaffold-inspect` and `run-inspect` skills

## Variables

**Independent Variables:**
1. **Training Dataset:** 5-letter words vs 9-letter words
2. **Evaluation Dataset:** 5-letter words vs 9-letter words

**Dependent Variable:**
- Capitalization accuracy (exact match)

**Controlled Variables:**
- Model: Llama-3.2-1B-Instruct
- LoRA rank: 64
- Training epochs: 2
- Training samples: 800 (per dataset)
- Capitalization ratio: 80%
- System prompt: "" (blank)

| Factor | Levels |
|--------|--------|
| Training Dataset | 5-letter words, 9-letter words, none (base) |
| Evaluation Dataset | 5-letter words, 9-letter words |

## All Runs

| Run Name | Model | Training Dataset | LoRA Rank | Epochs | Batch Size | Type | Est. Time |
|----------|-------|-----------------|-----------|--------|------------|------|-----------|
| train5L_rank64 | Llama-3.2-1B-Instruct | words_5L_80P_1000 (800 samples) | 64 | 2 | 4 | Fine-tuned | ~3 min |
| train9L_rank64 | Llama-3.2-1B-Instruct | words_9L_80P_1000 (800 samples) | 64 | 2 | 4 | Fine-tuned | ~3 min |
| base_control | Llama-3.2-1B-Instruct | - | - | - | - | Control | N/A |

**Notes:**
- **Type:** "Fine-tuned" requires training, "Control" evaluates base model only
- **Batch size:** Estimated at 4, will be auto-calculated during scaffolding
- Both fine-tuning runs can be submitted in parallel

## Evaluation Plan

Each run will be evaluated on BOTH datasets to test cross-length generalization:

### Evaluation Matrix

| Run Name | Dataset: 5-letter words | Dataset: 9-letter words | Epochs Evaluated | Notes |
|----------|------------------------|------------------------|------------------|-------|
| train5L_rank64 | ✓ In-distribution | ✓ Out-of-distribution | epoch_1 (final) | Tests if 5L training generalizes to 9L |
| train9L_rank64 | ✓ Out-of-distribution | ✓ In-distribution | epoch_1 (final) | Tests if 9L training generalizes to 5L |
| base_control | ✓ Baseline | ✓ Baseline | N/A | Establishes baseline performance |

**Key Comparisons:**

1. **In-distribution performance:**
   - train5L_rank64 on 5L dataset (should be high)
   - train9L_rank64 on 9L dataset (should be high)

2. **Out-of-distribution generalization:**
   - train5L_rank64 on 9L dataset (tests 5L → 9L generalization)
   - train9L_rank64 on 5L dataset (tests 9L → 5L generalization)

3. **Effect of fine-tuning:**
   - Compare each fine-tuned model to base_control on both datasets

4. **Symmetry of generalization:**
   - Is 5L → 9L generalization similar to 9L → 5L generalization?

### Evaluation Tasks

| Task Name | Script | Dataset | Description |
|-----------|--------|---------|-------------|
| cap_5L | `/home/mjs3/cruijff_kit/experiments/capitalization/inspect_task_capitalization.py` | words_5L_80P_1000.json | Tests capitalization on 5-letter words |
| cap_9L | `/home/mjs3/cruijff_kit/experiments/capitalization/inspect_task_capitalization.py` | words_9L_80P_1000.json | Tests capitalization on 9-letter words |

**Note:** The same inspect task script will be used with different datasets specified at evaluation time.

**Total Evaluations:** 6 (3 runs × 2 datasets)
**Estimated Evaluation Time:** ~6-12 minutes total (~1-2 min per evaluation)

## Resources

### Models

- **Llama-3.2-1B-Instruct**
  - Path: `/scratch/gpfs/MSALGANIK/pretrained-llms/Llama-3.2-1B-Instruct`
  - Verified: ✓ (2025-11-16)
  - Size: ~2.5 GB
  - Files: config.json, model.safetensors, tokenizer files

### Datasets

**Training Datasets:**

- **5-letter words (words_5L_80P_1000.json)**
  - Path: `/home/mjs3/cruijff_kit/data/green/capitalization/words_5L_80P_1000.json`
  - Format: JSON (instruct_dataset format)
  - Size: 60KB
  - Splits: train (800 samples), validation (100 samples), test (100 samples)
  - Capitalization: 80% capitalized, 20% lowercase
  - Verified: ✓ (2025-11-16, generated during experiment design)

- **9-letter words (words_9L_80P_1000.json)**
  - Path: `/home/mjs3/cruijff_kit/data/green/capitalization/words_9L_80P_1000.json`
  - Format: JSON (instruct_dataset format)
  - Size: 68KB
  - Splits: train (800 samples), validation (100 samples), test (100 samples)
  - Capitalization: 80% capitalized, 20% lowercase
  - Verified: ✓ (2025-11-16, generated during experiment design)

**Evaluation Datasets:**
- Same as training datasets (using test splits for evaluation)
- Each model will be evaluated on both 5L and 9L test sets

### Evaluation Tasks

- **Script:** `/home/mjs3/cruijff_kit/experiments/capitalization/inspect_task_capitalization.py`
  - Verified: ✓ (2025-11-16)
  - Description: Tests word capitalization accuracy using inspect-ai
  - Scorer: exact_match

### Disk Space

- **Available:** 9.0T on `/scratch/gpfs/`
- **Required:** ~15 GiB
  - Breakdown: 2 runs × 2 epochs × ~2.5 GiB per checkpoint + buffer
- **Status:** ✓ Sufficient space

## Configuration

### Training

- **Recipe:** lora_finetune_single_device
- **Epochs:** 2
- **LoRA Rank:** 64
- **LoRA Alpha:** 128 (2 × rank, default)
- **Learning Rate:** 3e-4 (default)
- **Batch Size:** 4 (estimated, will be auto-calculated based on GPU memory)
- **Dataset Packing:** Enabled (default)
- **Validation:** Yes (every 50 steps during training)
- **System Prompt:** "" (blank)
- **GPU:** 1 × 80GB GPU (constraint: gpu80)
- **Conda Environment:** ttenv

### Evaluation

- **Tool:** inspect-ai
- **Task:** capitalization
- **Temperature:** 0.0 (deterministic evaluation)
- **Scorer:** exact_match (case-sensitive string matching)
- **System Prompt:** "" (blank - must match training for valid comparison)
- **Max Samples:** Use full test set (100 samples per dataset)

### Output

- **Checkpoint Directory:** `/scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/`
- **Run Directory Pattern:** `{run_name}/` (e.g., `train5L_rank64/`, `train9L_rank64/`)
- **Checkpoint Pattern:** `{run_name}/epoch_{N}` (e.g., `train5L_rank64/epoch_0`, `train5L_rank64/epoch_1`)
- **Example Paths:**
  - `/scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/train5L_rank64/epoch_0`
  - `/scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/train5L_rank64/epoch_1`
  - `/scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/train9L_rank64/epoch_0`
  - `/scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/train9L_rank64/epoch_1`

**IMPORTANT:** System prompt must be identical between training and evaluation for valid comparisons.

## Compute Estimates

### Training Time

**Based on prior run:** `/scratch/gpfs/MSALGANIK/mjs3/cap_8L_llama32_lora_comparison_2025-10-18/1B_rank64/`

**Training Speed:**
- Observed: ~10 it/s (iterations per second)
- Model: Llama-3.2-1B-Instruct
- LoRA Rank: 64
- Batch Size: 4

**Calculations:**
- Dataset size: 800 samples
- Batch size: 4
- Steps per epoch: 800 / 4 = 200 steps
- Time per epoch (pure training): 200 steps / 10 it/s = 20 seconds
- Validation overhead: ~20 seconds per epoch
- **Total per epoch: ~40 seconds**
- **Total for 2 epochs: ~80 seconds ≈ 1.5 minutes per run**

**Conservative estimate with overhead:** ~3 minutes per run

**Total training time:**
- 2 runs × 3 minutes = 6 minutes (if run sequentially)
- **Can be parallelized:** Both runs submitted simultaneously → ~3 minutes wall time

### Evaluation Time

**Estimate:**
- Inference is ~2-3× faster than training
- ~1-2 minutes per evaluation (100 samples per test set)
- 6 total evaluations (3 runs × 2 datasets)
- **Total: ~6-12 minutes**

### Disk Space

**Checkpoint Size:**
- Based on prior run: 2.5 GiB per epoch (Llama-3.2-1B, LoRA rank 64)

**Total Storage:**
- Run 1 (train5L_rank64): 2 epochs × 2.5 GiB = 5 GiB
- Run 2 (train9L_rank64): 2 epochs × 2.5 GiB = 5 GiB
- Buffer (logs, intermediate files): ~5 GiB
- **Total: ~15 GiB**

### GPU Hours

**Training:**
- 2 runs × 3 minutes = 6 minutes = 0.1 GPU hours

**Evaluation:**
- 6 evaluations × 2 minutes = 12 minutes = 0.2 GPU hours

**Total: ~0.3 GPU hours**

## Naming Conventions

### Experiment Name

`cap_generalization_5L_9L_2025-11-16`

**Convention:**
- Task: `cap` (capitalization)
- Key factor: `generalization` (testing cross-length generalization)
- Variables: `5L_9L` (5-letter and 9-letter words)
- Date: `2025-11-16` (YYYY-MM-DD)

### Run Names

- **Format:** `train{word_length}_rank{lora_rank}`
- **Examples:**
  - `train5L_rank64` - Trained on 5-letter words, LoRA rank 64
  - `train9L_rank64` - Trained on 9-letter words, LoRA rank 64
  - `base_control` - Base model (no fine-tuning)

### Directory Structure

```
/scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/
├── experiment_summary.md (this file)
├── design-experiment.log
├── train5L_rank64/
│   ├── finetune.yaml
│   ├── finetune.slurm
│   ├── slurm-*.out
│   ├── epoch_0/
│   └── epoch_1/
├── train9L_rank64/
│   ├── finetune.yaml
│   ├── finetune.slurm
│   ├── slurm-*.out
│   ├── epoch_0/
│   └── epoch_1/
└── evaluations/
    ├── train5L_rank64_on_5L/
    ├── train5L_rank64_on_9L/
    ├── train9L_rank64_on_5L/
    ├── train9L_rank64_on_9L/
    ├── base_control_on_5L/
    └── base_control_on_9L/
```

## Quick Reference

### Paths

**Experiment Directory:**
```
/scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/
```

**Models:**
```
/scratch/gpfs/MSALGANIK/pretrained-llms/Llama-3.2-1B-Instruct
```

**Datasets:**
```
/home/mjs3/cruijff_kit/data/green/capitalization/words_5L_80P_1000.json
/home/mjs3/cruijff_kit/data/green/capitalization/words_9L_80P_1000.json
```

**Evaluation Script:**
```
/home/mjs3/cruijff_kit/experiments/capitalization/inspect_task_capitalization.py
```

### Common Commands

**List available models:**
```bash
ls /scratch/gpfs/MSALGANIK/pretrained-llms/
```

**Check datasets:**
```bash
ls -lh /home/mjs3/cruijff_kit/data/green/capitalization/*.json
```

**Find prior runs for estimate reference:**
```bash
find /scratch/gpfs/MSALGANIK/mjs3 -name "slurm-*.out" -path "*1B_rank64*" -size +50k | head -5
```

**Extract training speed from prior run:**
```bash
grep -E "[0-9.]+it/s" /path/to/slurm-*.out | tail -20
```

**Check disk space:**
```bash
df -h /scratch/gpfs/MSALGANIK/mjs3
```

**Check experiment directory:**
```bash
ls -lh /scratch/gpfs/MSALGANIK/mjs3/cap_generalization_5L_9L_2025-11-16/
```

### Next Steps

**Recommended Workflow:**

1. **Scaffold the experiment** (generates configs and SLURM scripts):
   ```
   Use scaffold-experiment skill
   ```

2. **Submit fine-tuning jobs:**
   ```
   Use run-experiment skill
   ```

3. **Monitor training:**
   ```bash
   squeue -u mjs3
   watch -n 10 squeue -u mjs3
   ```

4. **Evaluate models** (after training completes):
   ```
   run-experiment skill will handle this automatically
   ```

5. **Analyze results:**
   ```bash
   inspect view --port=$(get_free_port)
   ```
   Or use analyze-experiment skill (when available)

**Manual Workflow** (if needed):

1. Generate fine-tuning configs for each run using `tools/torchtune/setup_finetune.py`
2. Submit SLURM jobs: `sbatch finetune.slurm`
3. Generate evaluation configs using `tools/inspect/setup_inspect.py`
4. Submit evaluation jobs: `sbatch inspect.slurm`
5. View results: `inspect view`

## Expected Results

### Success Criteria

1. **Training converges:** Validation loss decreases across epochs for both runs
2. **High in-distribution accuracy:**
   - train5L_rank64 achieves >90% accuracy on 5L test set
   - train9L_rank64 achieves >90% accuracy on 9L test set
3. **Measurable generalization:** Out-of-distribution performance significantly better than base model

### Possible Outcomes

**Scenario 1: Strong Generalization**
- Both models achieve high accuracy on both in-distribution and out-of-distribution datasets
- Suggests the model learns a general capitalization rule independent of word length

**Scenario 2: Weak Generalization**
- Models perform well in-distribution but poorly out-of-distribution (similar to base model)
- Suggests the model memorizes length-specific patterns rather than learning a general rule

**Scenario 3: Asymmetric Generalization**
- 5L → 9L generalization differs from 9L → 5L
- Could indicate that training on shorter vs longer words has different learning dynamics

**Scenario 4: No Learning**
- Models don't improve over base model even in-distribution
- Suggests task is too difficult, needs more training, or configuration issues

## Notes

- Both datasets generated during experiment design (2025-11-16)
- Datasets are in the green tier (safe for AI assistance)
- This is a research experiment (not a sanity check)
- Results will inform whether capitalization tasks require length-specific training or if general patterns can be learned
- Consider follow-up experiments with intermediate lengths (6L, 7L, 8L) if interesting patterns emerge

---

**Experiment designed by:** Claude Code (design-experiment skill)
**Date created:** 2025-11-16
**Status:** Ready for scaffolding

**Comment by msalganik** [comment]
@niznik-dev who is this file for?  It seems to combine information for a human and for claude.  Maybe split into two?
